data/scenarios/Challenges/hanoi.yaml

```diff
@@ -0,0 +1,91 @@
+name: Towers of Hanoi
+description: XXX
+objectives:
+  - goal:
+      - Move all the disks from blah to blah blah
+    condition: |
+      try {
+        teleport self (2,-1);
+        x <- ishere "one";
+        teleport self (2,-2);
+        y <- ishere "blocked two";
+        teleport self (2,-3);
+        z <- ishere "blocked three";
+        return (x && y && z)
+      } { return false }
```

Comment:
Weird, if I factor it out the checking to a system robot, it works fine. :exploding_head:

Reply:
I just fixed the exception printing and got this:
![image](https://user-images.githubusercontent.com/44544735/189519180-c67fe8f7-1940-4fe2-88b0-b73704c82d5e.png)

Reply:
#675 should fix this. Once that is merged, this PR needs to be rebased on `main` and the tests should pass.

Reply:
I could have sworn that I had other objectives specified using `teleport self` before... but maybe not.